presto-main/src/main/java/io/prestosql/server/RangeJsonSerde.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.predicate.Marker;
+import io.prestosql.spi.predicate.Range;
+import io.prestosql.spi.type.Type;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import static io.prestosql.spi.predicate.Marker.Bound.EXACTLY;
+
+public class RangeJsonSerde
+{
+    private RangeJsonSerde() {}
+
+    public static class Serializer
+            extends JsonSerializer<Range>
+    {
+        @Override
+        public void serialize(Range range, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+                throws IOException
+        {
+            jsonGenerator.writeNumber(range.isSingleValue() ? 1 : 0);
+            if (range.isSingleValue()) {
+                serializerProvider.findValueSerializer(Type.class).serialize(range.getType(), jsonGenerator, serializerProvider);
+                serializerProvider.findValueSerializer(Block.class).serialize(range.getLow().getValueBlock().get(), jsonGenerator, serializerProvider);
+                return;
+            }
+
+            JsonSerializer<Object> markerSerializer = serializerProvider.findValueSerializer(Marker.class);
+            markerSerializer.serialize(range.getLow(), jsonGenerator, serializerProvider);
+            markerSerializer.serialize(range.getHigh(), jsonGenerator, serializerProvider);
```

Comment:
We should add a test in TestRangeJsonSerde verifying the Range class has exactly this two fields.
otherwise one can add a new optional field to Range, not knowing the dedicated serialization existss
the serialization test will still be passing, while it will be no longer correct

Reply:
Same applies to `Marker` class. As `if (range.isSingleValue()) {` branch essentially also provides specific serialiser for `Marker` objects (at least some subclass of those)

Reply:
I've added another test case with unbounded range.
Tests do round robin serialization, but also construct objects using `new Range, new Marker` so any changes to these objects should cause compilation to fail